paimon-python/pypaimon/pvfs/__init__.py

```diff
@@ -0,0 +1,769 @@
+        raise Exception(
+            f"Mv is not supported for path: {path1} to path: {path2}"
+        )
+
+    def lazy_load_class(self, module_name, class_name):
+        module = importlib.import_module(module_name)
+        return getattr(module, class_name)
+
+    def rm(self, path, recursive=False, maxdepth=None):
+        pvfs_identifier = self._extract_pvfs_identifier(path)
+        if isinstance(pvfs_identifier, PVFSDatabaseIdentifier):
+            database_name = pvfs_identifier.name
+            if recursive:
+                for table_name in self.rest_api.list_tables(database_name):
+                    self.rest_api.drop_table(Identifier.create(database_name, table_name))
+            self.rest_api.drop_database(database_name)
+            return True
+        elif isinstance(pvfs_identifier, PVFSTableIdentifier):
+            table_identifier = Identifier.create(pvfs_identifier.database, pvfs_identifier.name)
+            table = self.rest_api.get_table(table_identifier)
+            if pvfs_identifier.sub_path is None:
+                self.rest_api.drop_table(table_identifier)
+                return True
+            storage_type = self._get_storage_type(table.path)
+            storage_location = table.path
+            actual_path = pvfs_identifier.get_actual_path(storage_location)
+            fs = self._get_filesystem(pvfs_identifier, storage_type)
+            return fs.rm(
+                self._strip_storage_protocol(storage_type, actual_path),
+                recursive,
+                maxdepth,
+            )
+        raise Exception(
+            f"Rm is not supported for path: {path}."
+        )
+
+    def rm_file(self, path):
+        pvfs_identifier = self._extract_pvfs_identifier(path)
+        if isinstance(pvfs_identifier, PVFSTableIdentifier):
+            table_identifier = Identifier.create(pvfs_identifier.database, pvfs_identifier.name)
+            table = self.rest_api.get_table(table_identifier)
+            if pvfs_identifier.sub_path is not None:
+                storage_type = self._get_storage_type(table.path)
+                storage_location = table.path
+                actual_path = pvfs_identifier.get_actual_path(storage_location)
+                fs = self._get_filesystem(pvfs_identifier, storage_type)
+                return fs.rm_file(
+                    self._strip_storage_protocol(storage_type, actual_path),
+                )
+        raise Exception(
+            f"Rm file is not supported for path: {path}."
+        )
+
+    def rmdir(self, path):
+        pvfs_identifier = self._extract_pvfs_identifier(path)
+        if isinstance(pvfs_identifier, PVFSDatabaseIdentifier):
+            database_name = pvfs_identifier.name
+            for table_name in self.rest_api.list_tables(database_name):
+                self.rest_api.drop_table(Identifier.create(database_name, table_name))
+            self.rest_api.drop_database(database_name)
+            return True
```

Comment:
It seems rmdir should only support delete empty dir. Can we make a double check of its behavior?